### PR TITLE
group required, contract constructor, dependency

### PR DESCRIPTION
**Pull request by fudorec**

### CODE REVIEW

File: contracts/core/DeCusSystem.sol [hunk 1]
```diff
@@ -26,16 +26,22 @@ contract DeCusSystem is AccessControl, Pausable {
...
+    // const
     uint256 public constant MINT_REQUEST_GRACE_PERIOD = 24 hours;
+    bytes32 public constant GROUP_ADMIN_ROLE = keccak256("GROUP_ADMIN_ROLE");
 
     EBTC public ebtc;
     GroupRegistry public groupRegistry;
     KeeperRegistry public keeperRegistry;
     ReceiptController public receiptController;
-    SignatureValidator public signatureValidator;
+    SignatureValidator signatureValidator;
```
Comment (bill-clippy) [inline contracts/core/DeCusSystem.sol:37]: Why no let `DeCusSystem` inherit `SignatureValidator`?

File: contracts/core/DeCusSystem.sol [hunk 2]
```diff
@@ -26,16 +26,22 @@ contract DeCusSystem is AccessControl, Pausable {
...
 
-    constructor(address _admin) public {
-        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+    constructor() public {
+        _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
+
+        _setRoleAdmin(GROUP_ADMIN_ROLE, DEFAULT_ADMIN_ROLE);
```
Comment (bill-clippy) [inline contracts/core/DeCusSystem.sol:42]: No need to explictly set up `DEFAULT_ADMIN_ROLE`, as it is already the default admin for any new role. See [AccessControl.sol](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/378531b70906c906461fa14e5631f0569b938257/contracts/access/AccessControl.sol#L65)

File: contracts/keeper/GroupRegistry.sol [hunk 3]
```diff
@@ -25,13 +31,11 @@ contract GroupRegistry is AccessControl {
     GroupLib.GroupMap groups;
     mapping(string => uint256) address2id; // id starts from 1
 
-    constructor(address admin, address group_admin) public {
-        _setupRole(DEFAULT_ADMIN_ROLE, admin);
+    constructor() public {
+        _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
 
         _setRoleAdmin(GROUP_ADMIN_ROLE, DEFAULT_ADMIN_ROLE);
```
Comment (bill-clippy) [inline contracts/keeper/GroupRegistry.sol:37]: See comment above about `DEFAULT_ADMIN_ROLE`

File: contracts/keeper/KeeperRegistry.sol [hunk 4]
```diff
@@ -47,12 +47,10 @@ contract KeeperRegistry is AccessControl, IKeeperImport {
     }
 
     // write func
-    constructor(address admin, address keeper_admin) public {
-        _setupRole(DEFAULT_ADMIN_ROLE, admin);
+    constructor() public {
+        _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
 
         _setRoleAdmin(KEEPER_ADMIN_ROLE, DEFAULT_ADMIN_ROLE);
```
Comment (bill-clippy) [inline contracts/keeper/KeeperRegistry.sol:53]: See comment above about `DEFAULT_ADMIN_ROLE`

File: contracts/tokens/DeCus.sol [hunk 5]
```diff
@@ -11,8 +11,10 @@ contract DeCus is ERC20, AccessControl {
 
     bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
 
-    constructor(address minter) public ERC20(_name, _symbol) {
-        _setupRole(MINTER_ROLE, minter);
+    constructor() public ERC20(_name, _symbol) {
+        _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
+
+        _setRoleAdmin(MINTER_ROLE, DEFAULT_ADMIN_ROLE);
```
Comment (bill-clippy) [inline contracts/tokens/DeCus.sol:17]: See comment above about `DEFAULT_ADMIN_ROLE`

File: contracts/tokens/EBTC.sol [hunk 6]
```diff
@@ -14,11 +14,11 @@ contract EBTC is ERC20Burnable, ERC20Pausable, AccessControl {
...
     bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
 
-    constructor(address admin, address minter) public ERC20(_name, _symbol) {
-        _setupRole(DEFAULT_ADMIN_ROLE, admin);
+    constructor() public ERC20(_name, _symbol) {
+        _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
 
-        _setupRole(MINTER_ROLE, minter);
-        _setupRole(PAUSER_ROLE, admin);
+        _setRoleAdmin(MINTER_ROLE, DEFAULT_ADMIN_ROLE);
```
Comment (bill-clippy) [inline contracts/tokens/EBTC.sol:20]: See comment above about `DEFAULT_ADMIN_ROLE`

**Review by bill-clippy — CHANGES_REQUESTED** [review]

File: contracts/core/DeCusSystem.sol (hunk 1 above)
Comment (fudorec) [reply contracts/core/DeCusSystem.sol:37]: Next PR

**Pull request merged**